extensions/amp-image-slider/1.0/component.js

```diff
@@ -0,0 +1,607 @@
+  const stepExactCenter = useCallback(() => {
+    updatePositions(0.5);
+  }, [updatePositions]);
+
+  /**
+   * One step right
+   * @param {boolean=} opt_toEnd
+   * @private
+   */
+  const stepRight = useCallback(
+    (opt_toEnd) => {
+      // To the very end of right
+      if (opt_toEnd === true) {
+        updatePositions(1);
+      } else {
+        const newPercentage = limitPercentage(
+          getCurrentSliderPercentage() + stepSize
+        );
+        updatePositions(newPercentage);
+      }
+    },
+    [updatePositions, limitPercentage, getCurrentSliderPercentage, stepSize]
+  );
+
+  /**
+   * Handler on key down
+   * @param {Event} e
+   * @private
+   */
+  const onKeyDown = useCallback(
+    (e) => {
+      // Check if current element has focus
+      if (
+        doc.current.activeElement !== containerRef?.current &&
+        doc.current.activeElement !== containerRef?.current?./*OK*/ offsetParent
+      ) {
+        return;
+      }
+
+      animateHideHint();
+
+      switch (e.key.toLowerCase()) {
+        case 'left': // Edge non-standard, pre EdgeHTML 17
+        case 'arrowleft':
+          e.preventDefault();
+          e.stopPropagation();
+          stepLeft();
+          break;
+        case 'right': // Edge non-standard, pre EdgeHTML 17
+        case 'arrowright':
+          e.preventDefault();
+          e.stopPropagation();
+          stepRight();
+          break;
+        case 'pageup':
+          // prevent scrolling the page
+          e.preventDefault();
+          e.stopPropagation();
+          stepLeft(true);
+          break;
+        case 'pagedown':
+          e.preventDefault();
+          e.stopPropagation();
+          stepRight(true);
+          break;
+        case 'home':
+          e.preventDefault();
+          e.stopPropagation();
+          stepExactCenter();
+          break;
+      }
+    },
+    [animateHideHint, stepLeft, stepRight, stepExactCenter]
+  );
+  /**
+   * Handler on mouse move
+   * @param {Event} e
+   * @private
+   */
+  const onMouseMove = useCallback(
+    (e) => {
+      e.preventDefault();
+      pointerMoveX(e.pageX);
+    },
+    [pointerMoveX]
+  );
+
+  /**
+   * Handler on mouse button up
+   * @param {Event} e
+   * @private
+   */
+  const onMouseUp = useCallback(
+    (e) => {
+      e.preventDefault();
+      unlisten(unlistenMouseMove?.current);
+      unlisten(unlistenMouseUp?.current);
+    },
+    [unlisten]
+  );
+  /**
+   * Handler on mouse button down
+   * @param {Event} e
+   * @private
+   */
+  const onMouseDown = useCallback(
+    (e) => {
+      e.preventDefault();
+      pointerMoveX(e.pageX);
+
+      // In case, clear up remnants
+      // This is to prevent right mouse button down when left still down
+      unlisten(unlistenMouseMove?.current);
+      unlisten(unlistenMouseUp?.current);
+      unlistenMouseMove.current = listen(window, 'mousemove', onMouseMove);
+      unlistenMouseUp.current = listen(window, 'mouseup', onMouseUp);
```

Comment:
If you just instantly assigning the ref, why unlist it beforehand?

Reply:
@dethstrobe Please let us know if we are missing something.